syftlib/src/main/java/org/openmined/syft/Syft.kt

```diff
@@ -1,52 +1,38 @@
 package org.openmined.syft
 
 import io.reactivex.disposables.CompositeDisposable
-import kotlinx.serialization.ImplicitReflectionSerializer
-import kotlinx.serialization.json.JsonObject
-import kotlinx.serialization.json.JsonPrimitive
-import org.openmined.syft.domain.Protocol
-import org.openmined.syft.network.NetworkMessage
+import org.openmined.syft.network.CommunicationDataFactory
+import org.openmined.syft.network.REQUEST
 import org.openmined.syft.network.SignallingClient
 import org.openmined.syft.threading.ProcessSchedulers
 
 @ExperimentalUnsignedTypes
-class Syft constructor(
+class Syft private constructor(
     private val signallingClient: SignallingClient,
     private val schedulers: ProcessSchedulers
 ) {
+    private lateinit var syftWorker: Syft
 
-    private val compositeDisposable = CompositeDisposable()
-
-    @ImplicitReflectionSerializer
-    fun start() {
-
-        val disposable = signallingClient.start()
-                .map {
-                    when (it) {
-                        is NetworkMessage.SocketOpen -> {
-                            println("Socket open")
-                            send("And now that we are opened, I send a message")
-                        }
-                        is NetworkMessage.SocketClosed -> println("Socket was closed successfully")
-                        is NetworkMessage.SocketError -> println(it.throwable.message)
-                        is NetworkMessage.MessageReceived -> println(it)
-                        is NetworkMessage.MessageSent -> println("Message sent successfully")
-                    }
-                }
+    private val workerJobs = mutableListOf<Job>()
+    private val compositeDisposable = CompositeDisposable().add(
+        signallingClient.start()
                 .subscribeOn(schedulers.computeThreadScheduler)
                 .observeOn(schedulers.calleeThreadScheduler)
                 .subscribe()
+    )
 
-        compositeDisposable.add(disposable)
+    //TODO decide if we go by dependency injection who will initiate module
+    fun getSyftWorker(signallingClient: SignallingClient, schedulers: ProcessSchedulers): Syft {
+        if (!::syftWorker.isInitialized)
+            syftWorker = Syft(signallingClient, schedulers)
+
+        return syftWorker
     }
 
-    @ImplicitReflectionSerializer
-    fun send(message: String) = signallingClient.send(
-        "Personal", JsonObject(
-            mapOf(
-                "data" to JsonPrimitive(message),
-                "workerId" to JsonPrimitive("1234")
-            )
-        )
-    )
+    fun newJob(modelName: String, version: String): Job {
+        val job = Job(modelName, version)
+        signallingClient.send(REQUEST.JOB, CommunicationDataFactory.jobRequest(job))
```

Comment:
Just an idea:
What if we let the job do the `send`?
This way Syft could do a sequence like:
- create new job
- workerJobs.add(job)
- job.start()

and then `job.start()` would do the `send`

Reply:
job.start() will be called only after a successful validation at the pygrid and the by the syft instance which checks for connectivity, battery etc.
This sending is part of authentication which I believe should be the responsibility of syft

Reply:
Makes sense